lib/mix/tasks/phx.gen.release.ex

```diff
@@ -0,0 +1,141 @@
+defmodule Mix.Tasks.Phx.Gen.Release do
+  @shortdoc "Generates release files and optional Dockerfile for release-based deployments"
+
+  @moduledoc """
+  Generates release files and optional Dockerfile for release-based deployments.
+
+  The following release files are created:
+
+    * `lib/app_name/release.exs` - A release module containing tasks for running
+      migrations inside a release
+
+    * `rel/overlays/bin/migrate` - A migrate script for conveniently invoking
+      the release system migrations
+
+    * `rel/overlays/bin/server` - A server script for conveniently invoking
+      the release system with environment variables to start the phoenix web server
+
+  Note, the `rel/overlays` directory is copied into the release build by default when
+  running `mix release`.
+
+  When the `--docker` flag is passed, the following docker files are generated:
+
+    * `Dockerfile` - The Dockerfile for use in any standard docker deployment
+
+    * `.dockerignore` - A docker ignore file with standard elixir defaults
+
+  For extended release configuration, the `mix release.init`task can be used
+  in addition to this task. See the `Mix.Release` docs for more details.
+  """
+
+  use Mix.Task
+
+  @doc false
+  def run(args) do
+    docker? = "--docker" in args
+
+    if Mix.Project.umbrella?() do
+      Mix.raise("mix phx.gen.release is not supported in umbrella applications")
+    end
+
+    app = Mix.Phoenix.otp_app()
+    app_namespace = Mix.Phoenix.base()
+
+    binding = [
+      app_namespace: app_namespace,
+      otp_app: app
+    ]
+
+    Mix.Phoenix.copy_from(paths(), "priv/templates/phx.gen.release", binding, [
+      {:eex, "rel/migrate.sh.eex", "rel/overlays/bin/migrate"},
```

Comment:
Should we check if the user has Ecto? Maybe the app was installed with --no-ecto? 🤔

Reply:
yes, great catch!